projects/RabbitMQ.Client.OpenTelemetry/TraceProviderBuilderExtensions.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using OpenTelemetry.Context.Propagation;
+using RabbitMQ.Client;
+
+namespace OpenTelemetry.Trace
+{
+    public static class OpenTelemetryExtensions
+    {
+        public static TracerProviderBuilder AddRabbitMQInstrumentation(this TracerProviderBuilder builder)
+        {
+            RabbitMQActivitySource.UseRoutingKeyAsOperationName = true;
+            RabbitMQActivitySource.ContextExtractor = OpenTelemetryContextExtractor;
+            RabbitMQActivitySource.ContextInjector = OpenTelemetryContextInjector;
```

Comment:
Why we need this part? Can it be set by default in main package?

I would expect that this method will only call `builder.AddSource("RabbitMQ.Client.*")` as a convenient way to register it in TracerProviderBuilder. Other parts should be done in the instrumentation library.

Alternative options is to have here the second method:

```csharp
public static TracerProviderBuilder AddRabbitMQInstrumentation(this TracerProviderBuilder builder, Action<RabbitMQSpecificOptions>? configure);
```

where you can configure other methods externally.

Reply:
OpenTelemetryContextInjector is OTel specific for Baggage handling, that's pretty much the only reason for the package, because we don't want to pull OpenTelemetry.API dependencies into the main library.